modules/drivers/mongo/src/metabase/driver/mongo.clj

```diff
@@ -200,65 +192,45 @@
 
 (defmethod driver/describe-database :mongo
   [_ database]
-  (with-mongo-connection [^com.mongodb.DB conn database]
-    {:tables (set (for [collection (disj (mdb/get-collection-names conn) "system.indexes")]
+  (mongo.connection/with-mongo-database [^MongoDatabase db database]
+    ;; TODO: factor next line!
+    {:tables (set (for [collection (disj (into #{} (mongo.util/list-collection-names db)) "system.indexes")]
```

Comment:
I think `into` should rather be used when I do some transformation to the source collection. Thanks.

Here set operations are used just to remove one element from a set. Form in question was simplified, to use `(for [... :when...]` instead of set operations.